org.civicrm.reqangular/src/common/services/api/contract.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/apis',
+  'common/services/api'
+], function (apis, _) {
+  'use strict';
+
+  apis.factory('api.contract', ['$log', 'api', function ($log, api) {
+    $log.debug('api.contract');
+
+    return api.extend({
+
+      /**
+       * This method returns all the contracts.
+       *
+       * It chains an additional call to the `getfulldetails` endpoint to also return
+       * full details on contracts
+       *
+       * @param  {Object} params  matches the api endpoint params (ex. contact_id)
+       * @return {Promise}
+       */
+      all: function (params) {
+        $log.debug('ContractAPI.all');
+
+        params = params || {};
+
+        params['api.HRJobContract.getfulldetails'] = {'jobcontract_id': '$value.id'};
```

Comment:
>Also if you really need to getfulldetailscant you just call it directly? Because anyways you need to call 2nd API so better do it in the first place.

I need to get an object containing all the data I need in the most convenient way. Chaining is officially supported so I don't see a problem using it.